packages/presentation/presentation_publishing/interfaces/fetch/fetch.ts

```diff
@@ -0,0 +1,152 @@
+  timeslice: [number, number] | undefined;
+}
+
+function getFetchContext(api: unknown, isReload: boolean) {
+  const typeApi = api as Partial<
+    PublishesTimeRange & HasParentApi<Partial<PublishesUnifiedSearch & PublishesSearchSession>>
+  >;
+  return {
+    isReload,
+    filters: typeApi?.parentApi?.filters$?.value,
+    query: typeApi?.parentApi?.query$?.value,
+    searchSessionId: typeApi?.parentApi?.searchSessionId$?.value,
+    timeRange: typeApi?.timeRange$?.value ?? typeApi?.parentApi?.timeRange$?.value,
+    timeslice: typeApi?.timeRange$?.value ? undefined : typeApi?.parentApi?.timeslice$?.value,
+  };
+}
+
+function hasSearchSession(api: unknown) {
+  return apiHasParentApi(api) && apiPublishesSearchSession(api.parentApi)
+    ? typeof api.parentApi.searchSessionId$.value === 'string'
+    : false;
+}
+
+function hasLocalTimeRange(api: unknown) {
+  return apiPublishesTimeRange(api) ? typeof api.timeRange$.value === 'object' : false;
+}
+
+// Returns observables that emit to changes after subscribe
+// 1. Observables are not guaranteed to have an initial value (can not be used in combineLatest)
+// 2. Observables will not emit on subscribe
+function getBatchedObservables(api: unknown): Array<Observable<unknown>> {
+  const observables: Array<Observable<unknown>> = [];
+
+  if (apiPublishesTimeRange(api)) {
+    observables.push(api.timeRange$.pipe(skip(1)));
+  }
+
+  if (apiHasParentApi(api) && apiPublishesUnifiedSearch(api.parentApi)) {
+    observables.push(
+      combineLatest([api.parentApi.filters$, api.parentApi.query$]).pipe(
+        skip(1),
+        filter(() => !hasSearchSession(api))
+      )
+    );
+
+    if (apiHasParentApi(api) && apiPublishesTimeRange(api.parentApi)) {
+      const timeObservables: Array<Observable<unknown>> = [api.parentApi.timeRange$];
+      if (api.parentApi.timeslice$) {
+        timeObservables.push(api.parentApi.timeslice$);
+      }
+      observables.push(
+        combineLatest(timeObservables).pipe(
+          skip(1),
+          filter(() => !hasSearchSession(api) && !hasLocalTimeRange(api))
+        )
+      );
+    }
+  }
+
+  return observables;
+}
+
+// Returns observables that emit to changes after subscribe
+// 1. Observables are not guaranteed to have an initial value (can not be used in combineLatest)
+// 2. Observables will not emit on subscribe
+function getImmediateObservables(api: unknown): Array<Observable<unknown>> {
+  const observables: Array<Observable<unknown>> = [];
+  if (apiHasParentApi(api) && apiPublishesSearchSession(api.parentApi)) {
+    observables.push(api.parentApi.searchSessionId$.pipe(skip(1)));
+  }
+  if (apiHasParentApi(api) && apiPublishesReload(api.parentApi)) {
+    observables.push(api.parentApi.reload$.pipe(filter(() => !hasSearchSession(api))));
+  }
+  return observables;
+}
+
+export function fetch$(api: unknown): Observable<FetchContext> {
+  const onSubscribe$ = new BehaviorSubject<undefined>(undefined).pipe(first());
```

Comment:
Good call to separate out the `onSubscribe` value. I wonder if we could get the same behaviour with something like:

```typescript
merge(
    immediateChange$,
    batchedChanges$
 ).pipe(startWith(getFetchContext(api, false)));
```